app/src/main/java/com/ntg/user/mvpsample/tasks/TasksFragment.java

```diff
@@ -0,0 +1,227 @@
+        addTaskFab.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                presenter.addTask();
+            }
+        });
+
+    }
+
+    @Override
+    public View onCreateView(LayoutInflater inflater, ViewGroup container,
+                             Bundle savedInstanceState) {
+        View view = inflater.inflate(R.layout.fragment_tasks, container, false);
+        ButterKnife.bind(this, view);
+        llm = new LinearLayoutManager(getContext());
+        rvTasks.setLayoutManager(llm);
+        tasksAdapter = new TasksAdapter(new ArrayList<Task>(0), itemListener);
+        rvTasks.setAdapter(tasksAdapter);
+        return view;
+    }
+
+    @Override
+    public void onResume() {
+        super.onResume();
+        presenter.start();
+    }
+
+    @Override
+    public void setPresenter(TasksContract.Presenter presenter) {
+        this.presenter = presenter;
+    }
+
+    @Override
+    public void showTasks(List<Task> tasks) {
+        noTasksTxtView.setVisibility(View.GONE);
+        tasksAdapter.replaceData(tasks);
+    }
+
+    @Override
+    public void showNoTasks() {
+        noTasksTxtView.setVisibility(View.VISIBLE);
+    }
+
+    @Override
+    public void showAddNewTaskUI() {
+        Intent intent = new Intent(getContext(), AddTaskActivity.class);
+        startActivity(intent);
+    }
+
+    @Override
+    public void showTaskDetailsUI() {
+        Intent intent = new Intent(getContext(), TaskDetailsActivity.class);
+        startActivity(intent);
+    }
+
+    @Override
+    public void showTaskAsComplete() {
+
+    }
+
+    @Override
+    public void showTaskAsActive() {
+
+    }
+
+    TaskItemListener itemListener = new TaskItemListener() {
+        @Override
+        public void onTaskClick(Task clickedTask) {
+            Intent intent = new Intent(getContext(), TaskDetailsActivity.class);
+            intent.putExtra("task", clickedTask);
+            startActivity(intent);
+        }
+
+        @Override
+        public void onCompleteTaskClick(Task completedTask) {
+            presenter.updateTaskStatusAsCompleted(completedTask);
+        }
+
+        @Override
+        public void onActivateTaskClick(Task activatedTask) {
+            presenter.updateTaskStatusAsCompleted(activatedTask);
+        }
+    };
+
+    class TasksAdapter extends RecyclerView.Adapter<TasksAdapter.TasksViewHolder> {
+        List<Task> taskList;
+        TaskItemListener taskItemListener;
+
+        public TasksAdapter(List<Task> taskList, TaskItemListener taskItemListener) {
+            this.taskList = taskList;
+            this.taskItemListener = taskItemListener;
+        }
+
+        @Override
+        public TasksViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
+            View view = LayoutInflater.from(parent.getContext())
+                    .inflate(R.layout.rv_task_item, parent, false);
```

Comment:
rename rv_task_item to item_task
always refer to guidelines https://github.com/ribot/android-guidelines/blob/master/project_and_code_guidelines.md